backend/src/main/java/com/auction/auctionapp/rest/controller/ProductController.java

```diff
@@ -29,6 +29,54 @@
         return ResponseEntity.ok(new ArrayList<ProductResponse>(productService.getProductsByCategory(id)));
     }
 
+    @GetMapping("/products/range")
+    public ResponseEntity<List<ProductResponse>> getProductsInRange(@RequestParam(name = "low") double low,
+                                                                    @RequestParam(name = "high") double high) {
+        return ResponseEntity.ok(new ArrayList<ProductResponse>(productService.getProductsInRange(low, high)));
+    }
+
+    @GetMapping("/products/range-category")
+    public ResponseEntity<List<ProductResponse>> getProductsInRangeByCategory(@RequestParam(name = "productCategoryId") int id,
+                                                                              @RequestParam(name = "low") double low,
+                                                                              @RequestParam(name = "high") double high) {
+        return ResponseEntity.ok(new ArrayList<ProductResponse>(productService.getProductsInRangeByCategory(id, low, high)));
+    }
+
+    @GetMapping("/shop/sub")
+    public ResponseEntity<List<ProductResponse>> getProductsBySubCategory(@RequestParam(name = "subCategoryName") String name) {
+        return ResponseEntity.ok(new ArrayList<ProductResponse>(productService.getProductsBySubCategory(name)));
+    }
+
+    @GetMapping("/shop/sub-default")
+    public ResponseEntity<List<ProductResponse>> getProductsBySubCategoryDefaultSort(@RequestParam(name = "subCategoryName") String name) {
+        return ResponseEntity.ok(new ArrayList<ProductResponse>(productService.getProductsBySubCategoryDefaultSort(name)));
+    }
+
+    @GetMapping("/shop/sub-new")
+    public ResponseEntity<List<ProductResponse>> getProductsBySubCategoryNewArrivals(@RequestParam(name = "subCategoryName") String name) {
+        return ResponseEntity.ok(new ArrayList<ProductResponse>(productService.getProductsBySubCategoryNewArrivals(name)));
+    }
```

Comment:
it is better to send these filter properties in request body and on service have a logic which will determine which query will be executed, rather than having multiple endpoints